### PR TITLE
feat: function claim_rewards for claiming DeFi Spring rewards

### LINKED ISSUE #102: Claiming DeFI Spring rewards
We need to add a new functionality for claiming accumulated DeFI Spring rewards and then transferring them to some account for further processing. Information about issue is built from gathered info, so extra research may be needed.

Steps:
1. Read [guidelines](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5).
2. Add the new method `claim_rewards(claims, proofs)` to the `Deposit` smart contract
3. Check that position is open.
4. Claim all rewards one by one and transfer them to some address (any will do for now).
5. Test the code in the `test_loop.cairo` file

Definition of done:
Code is tested and works properly.

Please, apply only if you have already done at least a few tasks in Caior, also when you apply,  attach some link reference of cairo code.

### PR DESCRIPTION
**Pull request by SoarinSkySagar**
### Closes #102 

 - [x] Created ClaimData struct
 - [x] Written `claim_rewards` function
   - [x] Checking if position is open
   - [x] Claim all rewards one by one and transfer them to some address
 - [x] Written test for `claim_rewards`
 
### Tests Running
image

### CODE REVIEW

File: src/deposit.cairo [hunk 1]
```diff
@@ -325,6 +334,22 @@ mod Deposit {
                     }
                 );
         }
+
+        fn claim_rewards(ref self: ContractState, claim_data: Claim, proofs: Span<felt252>,) {
+            assert(self.is_position_open.read(), 'Position is not open');
+            assert(proofs.len() != 0, 'Proofs Span cannot be empty');
+
+            let airdrop_addr: ContractAddress =
```
Comment (faurdent) [inline src/deposit.cairo:342]: It was not by the start, but every week zkLend deploys a new contract and rewards might go there. Please, make this a parameter of `claim_rewards` method so it won't be constant.

File: src/interfaces.cairo [hunk 2]
```diff
@@ -38,3 +40,15 @@ pub trait IMarket<TContractState> {
     fn repay(ref self: TContractState, token: ContractAddress, amount: felt252);
     fn repay_all(ref self: TContractState, token: ContractAddress);
 }
+
+#[starknet::interface]
+pub trait IAirdrop<TContractState> {
```
Comment (faurdent) [inline src/interfaces.cairo:45]: Remove all unused methods from interface as they go in the line count for audit. Always try to keep as little code as possible.

File: src/types.cairo [hunk 3]
```diff
@@ -45,3 +45,17 @@ pub struct MarketReserveData {
     liquidation_bonus: felt252,
     debt_limit: felt252
 }
+
+#[derive(Copy, Drop, Serde)]
+pub struct Config {
```
Comment (faurdent) [inline src/types.cairo:50]: After a part of methods from `IAirdrop` is removed this one gonna be unneeded too.

File: src/deposit.cairo [hunk 4]
```diff
@@ -325,6 +334,22 @@ mod Deposit {
...
+                0x66cabe824da3ff583b967ce571d393e8b667b33415acc750397aa66b64a5a6c
+                .try_into()
+                .unwrap();
+            let airdrop_disp = IAirdropDispatcher { contract_address: airdrop_addr };
+
+            let res = airdrop_disp.claim(claim_data, proofs);
+            assert(res, 'Claim failed');
```
Comment (faurdent) [inline src/deposit.cairo:349]: When event removed make a one-liner.

File: tests/test_loop.cairo [hunk 5]
```diff
@@ -491,15 +493,105 @@ fn test_close_position_amounts_cleared() {
...
+            pool_key,
+            pool_price,
+            pool_price
+        );
+    stop_cheat_account_contract_address(deposit_disp.contract_address);
+
+    let initial_balance = strk_disp.balanceOf(user);
+    // println!("initial bal {}", initial_balance);
+
+    deposit_disp.claim_rewards(claim_data: claim_data, proofs: proofs);
```
Comment (faurdent) [inline tests/test_loop.cairo:587]: I think names are descriptive, why not to pass them as is

**Review by faurdent — CHANGES_REQUESTED** [review]
Some small changes required, mostly looks good

**Review by faurdent — APPROVED** [review]
@djeck1432 LGTM, approved

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Comment by SoarinSkySagar** [comment]
thank you @djeck1432 and @faurdent, would love to contribute to spotnet again 🚀

**Pull request merged**